File: 백준/Silver/10025. 게으른 백곰/게으른 백곰.py
n, k = map(int, input().split())

MAX_POS = 1000001
x = [0] * MAX_POS

for _ in range(n):
    g, i = map(int, input().split())
    x[i] += g  # 같은 위치에 여러 개 있을 수 있음

# 초기 윈도우 합 구하기
window_size = 2 * k + 1
current_sum = sum(x[:window_size])
max_ice = current_sum

# 슬라이딩 윈도우
for i in range(window_size, MAX_POS):
    current_sum = current_sum - x[i - window_size] + x[i]
    max_ice = max(max_ice, current_sum)

print(max_ice)

File: 백준/Silver/10025. 게으른 백곰/게으른 백곰.js
const input = require("fs").readFileSync("/dev/stdin", "utf-8").trim().split("\n");

const [n, k] = input[0].trim().split(" ").map(Number);
const MAXPOS = 1000001;
const x = new Array(MAXPOS).fill(0);

for (let j = 1; j < n + 1; j++) {
	const [g, i] = input[j].trim().split(" ").map(Number);
	x[i] = g;
}

const windowSize = 2 * k + 1; // 양쪽 k개, 나

let currentSum = x.slice(0, windowSize).reduce((a, b) => a + b, 0);
let maxIce = currentSum;

for (let i = windowSize; i < MAXPOS; i++) {
	currentSum = currentSum - x[i - windowSize] + x[i];
	maxIce = Math.max(maxIce, currentSum);
}

console.log(maxIce);
